iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/thread.hpp

```diff
@@ -28,10 +33,58 @@ constexpr uint64_t MAX_THREAD_NAME_LENGTH = 15U;
 
 using ThreadName_t = cxx::string<MAX_THREAD_NAME_LENGTH>;
 
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept;
-ThreadName_t getThreadName(pthread_t thread) noexcept;
+enum class ThreadError
+{
+    EMPTY_CALLABLE,
+    INSUFFICIENT_MEMORY,
+    INSUFFICIENT_PERMISSIONS,
+    INSUFFICIENT_RESOURCES,
+    INVALID_ATTRIBUTES,
+    UNDEFINED
+};
+
+class Thread
+{
+  public:
+    using callable_t = cxx::function<void()>;
```

Comment:
Since we use here an alias we can add this any time in the future with adjusting a single line. For the moment I would pursue the simplest approach and see how far we get with this.